### DISCUSSION TITLE
Build IM7 with fftw on macosx catalina with brew

### DISCUSSION THREAD

**Discussion by NcMichel** [ImageMagick/ImageMagick#2541] [Help]
Hello
I use this version on an Ubuntu machine
_Version: ImageMagick 7.0.10-29 Q16 x86_64 2020-09-10 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib cairo djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lcms lqr lzma openexr pangocairo png rsvg tiff webp wmf x xml zlib_

On my macosx catalina machine, I just didnt succeed in compiling with fftw, therefore I have to use
_Version: ImageMagick 7.0.10-29 Q16 x86_64 2020-09-10 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1) 
Delegates (built-in): bzlib freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib_

I buikd with brew, fftw is installed, and brew dont accept options now.
I modified the brew imagemagick.rb to obtain:
 _depends_on "pkg-config" => :build
  depends_on "freetype"
  depends_on "ghostscript"
  depends_on "jpeg"
  depends_on "libheif"
  depends_on "liblqr"
  depends_on "libomp"
  depends_on "libpng"
  depends_on "libtiff"
  depends_on "libtool"
  depends_on "little-cms2"
  depends_on "openexr"
  depends_on "openjpeg"
  depends_on "webp"
  depends_on "xz"
  depends_on "fftw"
  

  uses_from_macos "bzip2"
  uses_from_macos "libxml2"
  uses_from_macos "zlib"

  skip_clean :la

  def install
    # Avoid references to shim
    inreplace Dir["**/*-config.in"], "@PKG_CONFIG@", Formula["pkg-config"].opt_bin/"pkg-config"

    args = %W[
      --enable-osx-universal-binary=no
      --prefix=#{prefix}
      --disable-dependency-tracking
      --disable-silent-rules
      --disable-opencl
      --enable-shared
      --enable-static
      --with-freetype=yes
      --with-modules
      --with-openjp2
      --with-openexr
      --with-webp=yes
      --with-heic=yes
      --with-gslib
      --with-gs-font-dir=#{HOMEBREW_PREFIX}/share/ghostscript/fonts
      --with-lqr
      --with-fftw
      --without-pango
      --without-x_

-----------
Any idea what to do for obtaining fftw working on mac?
Thanks
Michel
======> UPDATE
rebuilding brew and writing .rb file again, then brew -s finally work

**_Version: ImageMagick 7.0.10-29 Q16 x86_64 2020-09-10 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1) 
Delegates (built-in): bzlib fftw freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps tiff webp xml zlib_**

**Comment by fmw42** [comment]
I  am on Mac OSX Sierra. I have been installing all my delegates from MacPorts and then install ImageMagick from source. See http://www.imagemagick.org/script/binary-releases.php and http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=29100 and http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=21502&p=88202&hilit=OSX+MacPorts#p88202

**Comment by Jmuccigr** [comment]
You can use another tap. I use this one and `--with-fftw --with-libheif --with-libwmf --with-x11`:

<https://github.com/blogabe/homebrew-xplanet>

**Reply by NcMichel** [reply]
oops. "invalid option --with-fftw"

**Reply by Jmuccigr** [reply]
This worked for me:

`brew install blogabe/xplanet/imagemagick --with-x11 --with-fftw --with-libwmf --with-libheif`

```
> magick -version
Version: ImageMagick 7.0.10-29 Q16 x86_64 2020-09-07 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1)
Delegates (built-in): bzlib fftw fontconfig freetype heic jng jp2 jpeg lcms ltdl lzma png tiff webp wmf x xml zlib
```

**Comment by rleaver152** [comment]
Just to add to @NcMichel very useful comments.    On a Mac (Monterey)  my process was:

```
cp /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/i/imagemagick.rb .  #copying to your current working directory
emacs imagemagick.rb # or editor of your choice
# amend the lines as he says, ie include depends_on "fftw" and --with-fftw
# then in your current working directory
brew uninstall --ignore-dependencies imagemagick
brew install --build-from-source imagemagick.rb
# ignore the warning "Warning: Treating imagemagick.rb as a formula." and any cask errors.
```

I hope this saves others some time.

**Reply by mrobe** [reply]
With some expert help, I was finally able to get this to work.

Attaching the edited `imagemagick.rb`, in case it might help others.

[imagemagick.rb.zip](https://github.com/user-attachments/files/17387790/imagemagick.rb.zip)
